libs/ledger-live-common/src/families/evm/synchronization.ts

```diff
@@ -68,33 +72,38 @@ export const getAccountShape: GetAccountShape = async (infos) => {
         return (acc?.blockHeight || 0) > (curr?.blockHeight || 0) ? acc : curr;
       }, null);
 
-  // This method could not be working if the integration doesn't have an API to retreive the operations
-  const lastCoinOperations = await (async () => {
-    try {
-      const { getLastCoinOperations } = getExplorerApi(currency);
-      return await getLastCoinOperations(
-        currency,
-        address,
-        accountId,
-        latestSyncedOperation?.blockHeight || 0
-      );
-    } catch (e) {
-      log("EVM Family", "Failed to get latest transactions", {
-        address,
-        currency,
-        error: e,
-      });
-      throw e;
-    }
-  })();
+  const { lastCoinOperations, lastTokenOperations, lastNftOperations } =
+    await (async () => {
+      try {
+        const { getLastOperations } = getExplorerApi(currency);
+        return await getLastOperations(
+          currency,
+          address,
+          accountId,
+          latestSyncedOperation?.blockHeight || 0
+        );
+      } catch (e) {
+        log("EVM Family", "Failed to get latest transactions", {
```

Comment:
We could unit test if the error is thrown when either getExplorerApi or getLastOperations fails

Reply:
It's covered here: https://github.com/LedgerHQ/ledger-live/blob/170bcd405fc999f25fc4cf8a1946a48cc53dc3aa/libs/ledger-live-common/src/families/evm/__tests__/unit/synchronization.unit.test.ts#L80

Reply:
Indeed my bad, this is the idea :) but then technically speaking it's the sub method that is failing ?